### DISCUSSION TITLE
Creating SDEF Intuitively

### DISCUSSION THREAD

**Discussion by SterlingButters** [idaholab/MontePy#357] [Ideas]
I see a lot of potential for this project in creating what would be otherwise complicated source definitions in MCNP. Without giving it too much thought at the moment, it would be nice to tell MontePy something to the effect of 

"I want 3 cylindrical volumetric sources each one with energy E1,E2,E3 with the source strength proportional to the cylinder's volume".
"I want 10 planar sources each oriented <some way> with energy E,..." 

The dependent source syntax has never been intuitive to specify (in my opinion). This project could bridge this gap.

**Comment by MicahGale** [comment]
That's a good idea, that we should keep in mind. I'm just really not looking forward to having implement the spider's web that will be needed for #22.
